DataManagerPlugin.cs

```diff
@@ -1,13 +1,167 @@
-using BepInEx;
+using Bep = BepInEx;
+using CG = System.Collections.Generic;
+using HL = HarmonyLib;
+using IO = System.IO;
+using Json = Newtonsoft.Json;
+using UE = UnityEngine;
 
 namespace Silksong.DataManager;
 
-[BepInAutoPlugin(id: "org.silksong-modding.datamanager")]
-public partial class DataManagerPlugin : BaseUnityPlugin
+[Bep.BepInAutoPlugin(id: "org.silksong-modding.datamanager")]
+public partial class DataManagerPlugin : Bep.BaseUnityPlugin
 {
-    private void Awake()
+    internal static DataManagerPlugin? Instance;
+
+    internal CG.List<(string GUID, IOnceSaveDataMod Mod)> onceSaveDataMods = new();
+
+    // We must use Start instead of Awake here - PluginInfos does not contain any mod instances
+    // when Awake runs.
+    private void Start()
+    {
+        Instance = this;
+        foreach (var (_, mod) in Bep.Bootstrap.Chainloader.PluginInfos)
+        {
+            if (mod.Instance is IOnceSaveDataMod modInstance)
+            {
+                Logger.LogInfo($"{mod.Metadata.GUID} uses once-save data");
+                onceSaveDataMods.Add((mod.Metadata.GUID, modInstance));
+            }
+        }
+        new HL.Harmony("org.silksong-modding.datamanager").PatchAll();
+    }
+
+    private static Json.JsonSerializerSettings jsonSettings = new()
```

Comment:
Should we include converters for common types such as vectors and keybinds? The latter may currently be in flux though

Reply:
In that case I propose we defer this for when that release is out, hopefully we'll know better what is or isn't needed (though we can add more at any time).